Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -127,6 +151,11 @@ struct JSONPackageCollectionProvider: PackageCollectionProvider {
                              readmeURL: package.readmeURL,
                              authors: nil)
             }
+
+            if !serializationOkay {
+                self.diagnosticsEngine.emit(warning: "Some collection information could not be deserialized likely due to invalid format, contact the collection's author (\(collection.generatedBy?.name ?? "n/a")) to address this issue.")
```

Comment:
Should the invalid payload get written to a temporary directory and referenced here so that it can be looked into by the author?

Reply:
in this case, the collection is coming from the remote URL where the author published it. we can add the URL tho

Reply:
Right, understood.  Ideally we'd have some way to attach the downloaded JSON to the log as an attachment that could be looked at later.  Perhaps writing it to a temporary location and providing a path to that is a bit too crufty, but I was suggesting it only as a workaround.

Reply:
that could be nice. I guess in this case the user can curl the URL if they really want to?

Reply:
That's a good point, although the data returned by querying the URL could have been changed in the meantime (or could have been truncated or corrupted during download even though the source was fine).  In any case, it's not needed for this PR, but as we continue to make the diagnostics from SwiftPM less opaque, I it would be good to eventually be able to retrieve the content that couldn't be parsed.  But not for this PR.

Reply:
totally agree